### ISSUE TITLE
CumulativeMs < (BeforeAllMs+BeforeEachMs+TestOnlyMs+AfterEachMs+AfterAllMs)  ?

### ISSUE THREAD

**Issue by vincent-fuchs** [grzegorz-aniol/junit5-watcher#6]
I was expecting CumulativeMs to be the sum of all the other numbers, but I see that regularly it's not the case and it's sometimes (largely) inferior :

![image](https://github.com/grzegorz-aniol/junit5-watcher/assets/9657583/23d9688e-11ee-4030-b2bc-053ad4744ff6)

how is the CumulativeMs number computed ?

**Comment by grzegorz-aniol** [comment]
@vincent-fuchs it's quite interesting, there should not be such differences.
Cumulative values is a difference between two timestamps, when test class was started and ended. To be more precise, the extension uses JUnit5 callbacks `BeforeAllCallback` and `AfterAllCallback`. For sure it shouldn't be higher than sum of individual metrics (before, test only, after). Maybe there is a bug. I will try to reproduce but if you can provide any simple class with this issue that would be nice.

**Comment by grzegorz-aniol** [comment]
Do you use a mixture of test classes with "without-nested" and with nested classes maybe? I didn't test this type of scenario yet.

**Comment by grzegorz-aniol** [comment]
I've just released 1.1, you may try to reproduce it, as it solves #2

**Comment by vincent-fuchs** [comment]
some findings on my side :

- for tests that use ` @Nested` , we get one row with only CumulativeMs for the top level, then one detailed row for each nesting level :
![image](https://github.com/grzegorz-aniol/junit5-watcher/assets/9657583/bbbe1977-93f6-47b4-8cb3-9a51745f6fc8)

The top level number is not the sum of the nested levels though. 

- probably something that is not supported yet, that gives strange numbers : some of my tests extend `org.junit.jupiter.api.extension.BeforeAllCallback` to perform some set up at a global level. is this supported ? it could explain the discrepancies I observe.

**Comment by grzegorz-aniol** [comment]
Can it be your custom extension is declared before `BenchmarkExtension` ?
E.g.
```
@ExtendWith(YourCustomExtension.class)
@ExtendWith(BenchmarkExtension.class)
class AbcTest {
...
}
```
In that can it may not detect/measure correctly all callbacks, as it needs to overlap all others, and be used as first extension in a chain by JUnit5. 
Also I found that AfterAll metric, for enclosing class, may be measured incorrectly when nested test is used. I will fix this, but I think it's not your case, as you don't use it.

**Comment by grzegorz-aniol** [comment]
Also, I found problem with Nested tests and how timestamps are collected in that case, I will work on it.

**Comment by grzegorz-aniol** [comment]
I improved the way how `junit5-watcher` handles nested tests. In addition, README contains more details about metrics, how they are calculated and how nested class metrics imply enclosing class metrics

**Comment by grzegorz-aniol** [comment]
Fix will be included in the next release, planned this week.

**Comment by grzegorz-aniol** [comment]
Released in 1.2
